internal/requester/get.go

```diff
@@ -1,16 +1,21 @@
 package requester
 
 import (
-	"encoding/json"
 	"fmt"
 	"io"
 	"net/http"
 	"reflect"
+
+	"code.cloudfoundry.org/jsonry"
 )
 
 func (r Requester) Get(url string, receiver any) error {
-	if reflect.TypeOf(receiver).Kind() != reflect.Ptr {
-		return fmt.Errorf("receiver must be of type Pointer")
+	receiverType := reflect.ValueOf(receiver)
+	switch {
+	case receiverType.Kind() != reflect.Ptr:
+		return fmt.Errorf("receiver must be a pointer to a struct, got non-pointer")
+	case receiverType.Elem().Kind() != reflect.Struct:
```

Comment:
thoughts: I am not sure if the method has the responsibility to decide what type of data is admitted when in the signature we use `any`. 
In general the use of reflect adds complexity and if it does not add significant value, I would avoid its use, but feel free to use it